File: solutions/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #Time O(N), Space O(1)
        left, right = 0, len(numbers) - 1
        val = numbers[left] + numbers[right]
        while val != target:
            if val < target:
                left += 1
            else:
                right -= 1
            val = numbers[left] + numbers[right]
        return [left+1, right+1]

File: solutions/167. Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0, j = numbers.size()-1;
        int total = numbers[i] + numbers[j];
        while (total != target) {
            if (total < target) {
                i++;
            }
            else {
                j--;
            }
            total = numbers[i] + numbers[j];
        }
        return {i+1, j+1};
    }
};
